### PR TITLE
Add ‘ghc_bindist’

### LINKED ISSUE #31: Support using official GHC bindists
#27 allows users to pull in their own tools to use during build. This lets them use GHC with prebuilt packages being available &c. The default there however uses `rules_nixpkgs`: in general we do not want to force nix on user.

We should provide basic targets for various GHC versions that will use official binary distribution. This will be plenty for users which only need vanilla GHC tooling to start building their project.

Likely related to #24

### PR DESCRIPTION
**Pull request by mrkkrp**
Close #31.

### CODE REVIEW

**Comment by mrkkrp** [comment]
@qnikst Suggested that we could download from gentoo repos, at least temporarily.

**Comment by Fuuzetsu** [comment]
> we somehow specify a shell script to execute on archive downloading

We can just use `build_file_contents` to perform arbitrary actions.

> we need to pass an absolute path to configure

Does this path matter after the installation is done? I'm guessing it does but if not then we can just give it the absolute path of target output directory and "install" into there.

>GHC also needs other binaries such as gcc to function properly

Unsure if this is a new problem. You still need this even today, the difference is that you just happen to make it available with GHC from your system (maybe?). So GCC can be a requirement. Actually C/C++ bazel rule fragments can be used for this probably like we do with `ld`.

>we create and host an archive with batteries included (with gcc and stuff). Then we just download that and we should be done

I think that's too much effort on our side, we'd have to do it for every platform and every release we want and it's much harder than adding a single line for a release.

File: haskell/ghc_bindist_build_template.bzl [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package(default_visibility = ["//visibility:public"])
```
Comment (mboes) [inline haskell/ghc_bindist_build_template.bzl]: This is not a Skylark source file, so should not have the `.bzl` file extension. These kings of templates are typically called `BUILD.ghc` or somesuch.

**Comment by mboes** [comment]
Alternate name for the rule: simply `ghc_bindist`.

**Comment by mrkkrp** [comment]
@mboes Your feedback has been addressed.

**Pull request merged**